delta-storage/src/test/java/io/cdap/delta/store/DraftServiceTest.java

```diff
@@ -0,0 +1,196 @@
+  }
+
+  @After
+  public void cleanupTest() throws Exception {
+    getStructuredTableAdmin().drop(DraftStore.TABLE_SPEC.getTableId());
+  }
+
+  @Test(expected = DraftNotFoundException.class)
+  public void testDraftNotFound() {
+    DraftService service = new DraftService(getTransactionRunner());
+    service.getDraft(new DraftId(new Namespace("ns", 0L), "testDraftNotFound"));
+  }
+
+  @Test(expected = InvalidDraftException.class)
+  public void testSaveInvalidDraftFails() {
+    DraftService service = new DraftService(getTransactionRunner());
+    Stage invalidSrc = new Stage("src",
+                                 new Plugin(null, DeltaSource.PLUGIN_TYPE, Collections.emptyMap(), Artifact.EMPTY));
+    Stage target = new Stage("t",
+                             new Plugin("oracle", DeltaTarget.PLUGIN_TYPE, Collections.emptyMap(), Artifact.EMPTY));
+    service.saveDraft(new DraftId(new Namespace("ns", 0L), "testSaveInvalidDraftFails"),
+                      new DeltaConfig(invalidSrc, target));
+  }
+
+  @Test
+  public void testTableListAndDetail() throws Exception {
+    DraftService service = new DraftService(getTransactionRunner());
+
+    DraftId draftId = new DraftId(new Namespace("ns", 0L), "testTableListAndDetail");
+    Stage src = new Stage("src",
+                          new Plugin("mock", DeltaSource.PLUGIN_TYPE, Collections.emptyMap(), Artifact.EMPTY));
+    Stage target = new Stage("t",
+                             new Plugin("oracle", DeltaTarget.PLUGIN_TYPE, Collections.emptyMap(), Artifact.EMPTY));
+    service.saveDraft(draftId, new DeltaConfig(src, target));
+
+    TableList expectedList = new TableList(Collections.singletonList(new TableSummary("deebee", "taybull", 3)));
+    List<ColumnDetail> columns = new ArrayList<>();
+    columns.add(new ColumnDetail("id", JDBCType.INTEGER, false));
+    columns.add(new ColumnDetail("name", JDBCType.VARCHAR, false));
+    columns.add(new ColumnDetail("age", JDBCType.INTEGER, true));
+    TableDetail expectedDetail = new TableDetail("deebee", "taybull", Collections.singletonList("id"), columns);
+    DeltaSource mockSource = new MockSource(expectedList, expectedDetail);
+    Configurer mockConfigurer = new MockConfigurer(mockSource, null);
+    Assert.assertEquals(expectedList, service.listDraftTables(draftId, mockConfigurer));
+    Assert.assertEquals(expectedDetail, service.describeDraftTable(draftId, mockConfigurer, "deebee", "taybull"));
+  }
+
+  @Test(expected = DraftNotFoundException.class)
+  public void testListTablesFromNonexistantDraft() throws IOException {
+    DraftService service = new DraftService(getTransactionRunner());
+    service.listDraftTables(new DraftId(new Namespace("ns", 0L), "testListTablesFromNonexistantDraft"),
+                            new MockConfigurer(null, null));
+  }
+
+  /**
+   * Mock source that returns pre-determined table list and table detail.
+   */
+  private static class MockSource implements DeltaSource {
```

Comment:
Do you think we can have this source as a separate class so other tests can use it?

Reply:
moved MockSource and MockConfigurer to their own classes